cardano-api/src/Cardano/Api/Alonzo/Render.hs

```diff
@@ -0,0 +1,73 @@
+module Cardano.Api.Alonzo.Render
+  ( renderScriptIntegrityHash
+  , renderScriptHash
+  , renderMissingRedeemers
+  , renderScriptPurpose
+  , renderBadInputsUTxOErr
+  , renderValueNotConservedErr
+  , renderTxId
+  ) where
+
+import           Cardano.Ledger.Crypto (StandardCrypto)
+import           Cardano.Ledger.Shelley.API hiding (ShelleyBasedEra)
+import           Cardano.Prelude
+import           Data.Aeson (ToJSON (..), Value(..), object, (.=))
+import           Data.Aeson.Types (Pair)
+import           Ouroboros.Consensus.Shelley.Ledger hiding (TxId)
+import           Prelude hiding ((.), map, show)
+
+import qualified Cardano.Api.Address as Api
+import qualified Cardano.Api.Certificate as Api
+import qualified Cardano.Api.Ledger.Mary as Api
+import qualified Cardano.Api.Script as Api
+import qualified Cardano.Api.SerialiseRaw as Api
+import qualified Cardano.Api.SerialiseTextEnvelope as Api
+import qualified Cardano.Api.TxBody as Api
+import qualified Cardano.Crypto.Hash.Class as Crypto
+import qualified Cardano.Ledger.Alonzo.Tx as Alonzo
+import qualified Cardano.Ledger.SafeHash as SafeHash
+import qualified Data.Aeson as Aeson
+import qualified Data.Aeson.Key as Aeson
+import qualified Data.Set as Set
+import qualified Ouroboros.Consensus.Ledger.SupportsMempool as Consensus
+
+renderScriptIntegrityHash :: Maybe (Alonzo.ScriptIntegrityHash StandardCrypto) -> Value
+renderScriptIntegrityHash (Just witPPDataHash) =
+  String . Crypto.hashToTextAsHex $ SafeHash.extractHash witPPDataHash
+renderScriptIntegrityHash Nothing = Aeson.Null
+
+renderScriptHash :: ScriptHash StandardCrypto -> Text
+renderScriptHash = Api.serialiseToRawBytesHexText . Api.fromShelleyScriptHash
+
+renderMissingRedeemers :: [(Alonzo.ScriptPurpose StandardCrypto, ScriptHash StandardCrypto)] -> Value
+renderMissingRedeemers scripts = object $ map renderTuple  scripts
+ where
+  renderTuple :: (Alonzo.ScriptPurpose StandardCrypto, ScriptHash StandardCrypto) -> Pair
+  renderTuple (scriptPurpose, sHash) =
+    Aeson.fromText (renderScriptHash sHash) .= renderScriptPurpose scriptPurpose
```

Comment:
So for example we don't need `renderScriptHash`:
```
-renderScriptHash :: ScriptHash StandardCrypto -> Text
-renderScriptHash = Api.serialiseToRawBytesHexText . Api.fromShelleyScriptHash
-
 renderMissingRedeemers :: [(Alonzo.ScriptPurpose StandardCrypto, ScriptHash StandardCrypto)] -> Value
 renderMissingRedeemers scripts = object $ map renderTuple  scripts
  where
   renderTuple :: (Alonzo.ScriptPurpose StandardCrypto, ScriptHash StandardCrypto) -> Pair
   renderTuple (scriptPurpose, sHash) =
-    Aeson.fromText (renderScriptHash sHash) .= renderScriptPurpose scriptPurpose
+    Aeson.fromText (Api.serialiseToRawBytesHexText $ Api.ScriptHash sHash) .= renderScriptPurpose scriptPurpose
 ```

Reply:
We shouldn't need to add a `Cardano.Api.Alonzo.Render` module

Reply:
It's duplicate, but they are inaccessible because they are in `cardano-node`.

Reply:
The `Render` module is internal, so I don't think it will be exposed by the API.

Reply:
This module is gone.